src/NostrIgnition.ts

```diff
@@ -239,93 +263,49 @@ const init = async (ignitionOptions: NostrIgnitionOptions) => {
-            if (nostrModalNpubOrNip05.value.length > 0) {
+        nostrModalBunkerInput.addEventListener("input", function () {
+            nostrModalBunkerInputError.innerText = "";
+            nostrModalBunkerInputError.style.display = "none";
+            nostrModalBunkerInput.classList.remove("invalid");
+            if (nostrModalBunkerInput.value.length > 0) {
                 nostrModalSignInSubmit.disabled = false;
             } else {
                 nostrModalSignInSubmit.disabled = true;
             }
         });
 
         // Add event listener for sign in form
-        nostrModalSignInSubmit.addEventListener("click", async function (event) {
+        nostrModalSignInForm.addEventListener("submit", async function (event) {
             event.preventDefault();
             nostrModalSignInSubmit.disabled = true;
             nostrModalSignInSubmitText.style.display = "none";
             nostrModalSignInSubmitSpinner.style.display = "block";
 
-            let remotePubkey: string | null = null;
-            // Order is important here. NIP05_REGEX is pretty loose so it will match an npub
-            if (NPUB_REGEX.test(nostrModalNpubOrNip05.value)) {
-                // Decode pubkey from npub
-                remotePubkey = decode(nostrModalNpubOrNip05.value).data as string;
-            } else if (PUBKEY_REGEX.test(nostrModalNpubOrNip05.value)) {
-                // Looks like a pubkey
-                remotePubkey = nostrModalNpubOrNip05.value;
-            } else if (NIP05_REGEX.test(nostrModalNpubOrNip05.value)) {
-                // Look up pubkey for nip05
-                const profilePointer = await queryProfile(nostrModalNpubOrNip05.value);
-                if (profilePointer) {
-                    remotePubkey = profilePointer.pubkey;
-                } else {
-                    nostrModalNpubOrNip05.setCustomValidity("Error fetching Pubkey from NIP-05 value.");
-                    nostrModalNpubOrNip05.classList.add("invalid");
-                    nostrModalNpubOrNip05Error.innerText = "Error fetching Pubkey from NIP-05 value.";
-                    nostrModalNpubOrNip05Error.style.display = "block";
-                    // Remove spinner and re-enable submit button
-                    nostrModalSignInSubmit.disabled = false;
-                    nostrModalSignInSubmitText.style.display = "block";
-                    nostrModalSignInSubmitSpinner.style.display = "none";
-                    return;
-                }
-            } else {
+            const bunkerPointer = await parseBunkerInput(nostrModalBunkerInput.value);
+            if (!bunkerPointer) {
                 // Nothing matches the value - it's an error.
-                nostrModalNpubOrNip05.setCustomValidity("Invalid Pubkey, npub, or NIP-05");
-                nostrModalNpubOrNip05.classList.add("invalid");
+                nostrModalBunkerInput.setCustomValidity("Invalid NIP-05 name@domain address or bunker:// URI.");
```

Comment:
Why remove the pubkey or npub options here?

Reply:
because that's a horrible antipattern. I think we must enforce specific relays to be passed through the nostr.json `nip46` attribute or the `bunker://` URL, otherwise everybody will just use some big central hardcoded relay and become dependent on it.

Reply:
For posterity, we talked out of band on this and agree the NIP46 stuff should only handle nip-05 or bunker URLs and ignition will add back a step to allow users to input an `npub` or `pubkey` value (which means using a bootstrap relay to fetch the user's profile data)